src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -1218,46 +1219,91 @@ namespace winrt::TerminalApp::implementation
-                // (or any other negative number) on the commandline. So the set
-                // of possible values here is {-1, 0, ℤ+}
-                return *parsedTarget;
-            }
-            else
+            const std::string parsedTarget{ appArgs.GetTargetWindow() };
+
+            // If the user did not provide any value on the commandline,
+            // then lookup our windowing behavior to determine what to do
+            // now.
+            if (parsedTarget.empty())
             {
-                // If the user did not provide any value on the commandline,
-                // then lookup our windowing behavior to determine what to do
-                // now.
+                int32_t windowId = WindowingBehaviorUseNew;
                 switch (windowingBehavior)
                 {
+                case WindowingMode::UseNew:
+                    windowId = WindowingBehaviorUseNew;
+                    break;
                 case WindowingMode::UseExisting:
-                    return WindowingBehaviorUseExisting;
+                    windowId = WindowingBehaviorUseExisting;
+                    break;
                 case WindowingMode::UseAnyExisting:
-                    return WindowingBehaviorUseAnyExisting;
-                case WindowingMode::UseNew:
-                default:
-                    return WindowingBehaviorUseNew;
+                    windowId = WindowingBehaviorUseAnyExisting;
+                    break;
+                }
+                return winrt::make<FindTargetWindowResult>(windowId, L"");
+            }
+
+            // Here, the user _has_ provided a window-id on the commandline.
+            // What is it? Let's start by checking if it's an int, for the
+            // window's ID:
+            try
+            {
+                int32_t windowId = ::base::saturated_cast<int32_t>(std::stoi(parsedTarget));
+
+                // If the user provides _any_ negative number, then treat it as
+                // -1, for "use a new window".
+                if (windowId < 0)
+                {
+                    windowId = -1;
+                }
+
+                // Hooray! This is a valid integer. The set of possible values
+                // here is {-1, 0, ℤ+}. Let's return that window ID.
+                return winrt::make<TerminalApp::implementation::FindTargetWindowResult>(windowId, L"");
+            }
+            catch (...)
+            {
+                // Value was not a valid int. It could be any other string to
+                // use as a title though!
+                //
+                // First, check the reserved keywords:
+                if (parsedTarget == "new")
+                {
+                    return winrt::make<FindTargetWindowResult>(WindowingBehaviorUseNew, L"");
+                }
+                else if (parsedTarget == "last")
+                {
+                    return winrt::make<FindTargetWindowResult>(WindowingBehaviorUseExisting, L"");

```

Comment:
```suggestion
                    return winrt::make<FindTargetWindowResult>(WindowingBehaviorUseExisting, L"last");
```
Should we send the right string too? Since they mean the same thing?